pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+        ALLOW_ANY = 0;
+        // Restrict outbound traffic to services defined in the service registry as well
+        // as those defined through ServiceEntries
+        REGISTRY_ONLY = 1;
+    }
+
+    // If set, allows traffic in cases when the mixer policy service cannot be reached.
+    google.protobuf.BoolValue policy_check_fail_open = 10;
+    // Selects the outbound traffic policy mode.
+    OutboundPolicy outbound_traffic_policy_mode = 11;
+
+    PolicyComponentSpec policy = 50;
+}
+
+// Configuration options for the telemetry feature.
+message TelemetryFeatureSpec {
+    // Selects whether telemetry is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all telemetry components are installed into.
+    string namespace = 2;
+
+    TelemetryComponentSpec telemetry = 51;
+}
+
+// Configuration options for security feature.
+message SecurityFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
+    google.protobuf.BoolValue enabled = 1;
+
+    // Namespace that security components are installed into.
+    string namespace = 2;
+
+    // Enables MTLS for control plane.
+    // Will result in delays starting the pods while secrets are propagated, not recommended for tests.
+    google.protobuf.BoolValue control_plane_mtls = 13;
+
+    // Enables strict MTLS for service to service traffic, or permissive mode MTLS if unset.
+    google.protobuf.BoolValue data_plane_mtls_strict = 14;
+
+    CitadelComponentSpec citadel = 20;
+    CertManagerComponentSpec cert_manager = 21;
+    NodeAgentComponentSpec node_agent = 22;
+}
+
+// Configuration options for configuration management feature.
+message ConfigManagementFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
+    google.protobuf.BoolValue enabled = 1;
+
+    // Namespace that security components are installed into.
+    string namespace = 2;
+
+    GalleyComponentSpec galley = 20;
+}
+
+// Configuration options for the pilot component.
+message PilotComponentSpec {
+    CommonComponentSpec common = 1;
+
+    // Enables sidecar in the pilot pod.
+    google.protobuf.BoolValue sidecar = 10;
+}
+
+// Configuration options for the proxy.
+message ProxyComponentSpec {
+    CommonComponentSpec common = 1;
+}
+
+// Configuration options for the sidecar injector component.
+message SidecarInjectorComponentSpec {
+    CommonComponentSpec common = 1;
+    // Selects whether auto-injection is enabled in namespaces by default.
+    google.protobuf.BoolValue enable_namespaces_by_default = 10;
+}
+
+// Configuration options for the policy enforcement component.
+message PolicyComponentSpec {
+    CommonComponentSpec common = 1;
+}
+
+// Configuration options for the telemetry component.
+message TelemetryComponentSpec {
+    CommonComponentSpec common = 1;
```

Comment:
I'm not sure if we need `loadshedding` in `TelemetryComponentSpec` , it's a telemetry specific parameter.

Reply:
do you mean we should add it? i guess i don't see which field your comment is talking about.

Reply:
PTAL: https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/values.yaml#L33-L37

Reply:
resolving for now, feel free to ping me or open an issue if there's still a concern.